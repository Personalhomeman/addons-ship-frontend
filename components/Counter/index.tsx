import { Component, Fragment } from 'react';

type CounterState = {
  count: number;
};

export default class Counter extends Component<{}, CounterState> {
  state: CounterState = {
    count: 0
  };

  increment = () => {
    const { count } = this.state;
    this.setState({ count: count + 1 });
  };

  render() {
    const { count } = this.state;

    return (
      <Fragment>
        <p>Current count: {count}</p>
        <button onClick={this.increment}>Increment</button>
      </Fragment>
    );
  }
}
