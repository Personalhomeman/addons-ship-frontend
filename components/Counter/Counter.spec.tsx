import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Counter from '.';

describe('Counter', () => {
  it('renders correctly', () => {
    const comp = shallow(<Counter />);
    expect(toJSON(comp)).toMatchSnapshot();
  });

  it('updates the counter', () => {
    const comp = shallow(<Counter />);

    const button = comp.find('button');
    button.simulate('click');

    expect(toJSON(comp)).toMatchSnapshot();
  });
});
