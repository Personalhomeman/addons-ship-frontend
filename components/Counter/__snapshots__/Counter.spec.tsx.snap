// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<Fragment>
  <p>
    Current count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</Fragment>
`;

exports[`Counter updates the counter 1`] = `
<Fragment>
  <p>
    Current count: 
    1
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</Fragment>
`;
